### DISCUSSION TITLE
Add support for frontmatter

### DISCUSSION THREAD

**Discussion by juanolon** [emad-elsaid/xlog#55] [Ideas]
i'm using zk for my zettelkasten, which save each note file using an id (https://github.com/zk-org/zk/blob/main/docs/note-id.md). The title of the note is set using YAML frontmatter. now, using xlog as my frontend, the note name is the id instead of the title key. It would be useful to support YAML frontmatter.
Currently, it could be use in two areas:
* title
* tags

**Comment by emad-elsaid** [comment]
yeah, we need to think deeply about this to make it work with what we have. like when a page file name is mentioned does it still link to it? or just the title in the frontmatter?

**Reply by juanolon** [reply]
that's true. I would say (and to keep it simple), the attributes in frontmatter doesn't affect the routing/serving of pages as they are used as metadata. I mean the key `title` would be only use for front matters ;) That would only affect the `title.html` template and how links are generated. I think, the later may be the big change?

This would mean, i have a file asd.md, with the `title` key set to "front matters". then the server will still serve the page `/asd` as usual, but the title on the page shows "front matters"

**Reply by emad-elsaid** [reply]
Simple enough. this one is easy to implement. Another issue is which value to use when auto linking. 
because most people will use it like so: 
- the file name: "about-me.md"
- the title "About me"
- Now in another file when you write "You can know more about me by reading pages on my website"
- if we're using the front-matter `title` as page title only. then `about me` in the previous text won't link. and only if the text says "You can ..... about-me by reading ...." to match the file name.
So I guess another decision we'll need to do is to use the title for auto-linking.

**Reply by juanolon** [reply]
How did it work until now? was only one word allowed? In any case, the file name could be `3fs546.md`, but the title `about me`, so there is no necessary "conversion" between file name and title.
I would say, support both, but take preference on the `title`. If no title was found, then we take the file path.

~We could think on adding a new field to the `Page` interface~. I just saw it, there is already `Name` and `FileName`, so `Name` could return the title from front-matter, whenever available, otherwise, the old value.

**Reply by juanolon** [reply]
btw. i started a new pull request with the changes :)
